### ISSUE TITLE
Export private IPs of NLBs

### ISSUE THREAD

**Issue by henrytk** [hashicorp/terraform-provider-aws#21893]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Context and use case

In https://github.com/hashicorp/terraform-provider-aws/issues/3007 it was requested that the private IPs were made available. That request was satisfied in https://github.com/hashicorp/terraform-provider-aws/pull/11404 for anyone wanting to specify which IPs they wanted to use.

This issue is to request access to the private IPs in the attributes of the resource, even when static IPs were not specified in the resource. It is a valid use case to want to delegate the selection of IPs to AWS, but the need to read those IPs for security groups still exists in said use case.

See below for expected and actual behaviour. I've tried running `terraform refresh`, but that doesn't help populate the attribute in the output.

### Terraform CLI and Terraform AWS Provider Version

Terraform 0.14.x and AWS 3.66.0

### Affected Resource(s)

* aws_lb

### Expected Behavior

When creating an NLB without specifying any `subnet_mapping`, I expect the properties to still include the `subnet_mapping` _and_ for it to include the private IP addresses of the interfaces in those subnets.

### Actual Behavior

If you do not specify a `subnet_mapping` the output includes the subnet IDs, but not the private IPs:

```
  "subnet_mapping" = toset([
    {
      "allocation_id" = ""
      "ipv6_address" = ""
      "outpost_id" = ""
      "private_ipv4_address" = ""
      "subnet_id" = "subnet-02614cbee7a0f4e7c"
    },
    {
      "allocation_id" = ""
      "ipv6_address" = ""
      "outpost_id" = ""
      "private_ipv4_address" = ""
      "subnet_id" = "subnet-097d908985e142bfc"
    },
  ])
  ```

### Steps to Reproduce

1. Create an aws_lb Network load balancer, specifying subnet IDs, not a subnet mapping
2. Output the aws_lb properties

**Comment by hannes-ucsc** [comment]
Yes, essentially this is the same as https://github.com/hashicorp/terraform-provider-aws/issues/3007 which was incorrectly declared closed. The alleged solution for that only implemented specifying the LB IPs as *arguments*, but it doesn't export them as *attributes*.

**Comment by mmerickel** [comment]
Ran into this as well... When client ip preservation is disabled AWS documents that these IPs can be used to further secure access to the targets so it'd be great if the attributes would be populated. Currently it returns an empty string.

**Comment by hannes-ucsc** [comment]
I don't understand the logic behind automatically closing issue if they receive no comments. This issue is clearly defined and reproducible. There is nothing else to do other than to solve it. This process of automatically closing issues only incentivizes users to make meaningless "yes, I need this too" comments that add no informational value other than preventing the automatic closure.

**Comment by kengibous** [comment]
This would be very helpful to be able to retrieve the allocated private ips when provisioning an NLB.
Explicitly providing IPs addresses instead of just providing subnets is more cumbersome.

Providing these details after creating the NLB would be very helpful - 

![Image](https://github.com/user-attachments/assets/5dbd76fd-92de-4c6b-9c1f-1fe96ec0a0f5)
